MapboxNavigation/NavigationMapView.swift

```diff
@@ -289,6 +295,11 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         if (tracksUserCourse) {
             updateCourseTracking(location: userLocationForCourseTracking, camera:self.camera, animated: false)
         }
+    
+        // TODO: Add style observation to be able to re-add highlighted building whenever it happens.
+        
+        // FIXME: Since building might be already highlighted and we do not have any info
+        // regarding its identification after style change such highlighted building will be lost.
```

Comment:
If the user approaches the destination at around twilight, the UI style could change from DayStyle to NightStyle, making the map style change as well, causing the building to go away. That could make the map less usable.

We could supplement RouteMapViewController’s style observer with an `NSKeyValueObserving.willChangeValue(forKey:)` implementation that stores the extrusion layer’s predicate and expressions to reapply them after the style change. Better yet, `NavigationViewController.updateMapStyle(_:animated:)` could handle reapplying the buildings.

Reply:
Thanks for idea. I'll create separate ticket to tackle this as a tail-work. More testing will be required to check how extrusion behaves when style was dynamically changed, that is the reason I've added such ability in sample [here](https://github.com/mapbox/navigation-ios-examples/pull/71).